kore/src/Kore/Reachability/Prove.hs

```diff
@@ -479,6 +490,16 @@ proveClaimStep _ stuckCheck claims axioms executionGraph node =
         | otherwise =
             transitionRule' stuckCheck claims axioms prim state
 
+    withTimeout execStep = case timeout of
+        Nothing -> Just <$> execStep
+        Just (StepTimeout st) -> do
+            let warnThread = liftIO . threadDelay $ st * 1000000
+            race warnThread execStep >>= \case
```

Comment:
Hmm, I think it's nice to have this behavior: if a step takes way too long it doesn't make sense for the user to wait until it actually finishes, because it might even be looping.

The only issue I can see here is with the `Simplifier` state, in order for this to be 100% correct we would also need to rollback any changes made to the state layer. I don't think this happens automatically, does it?

Now, that layer is a cache so maybe it's actually not problematic, but if we end up modifying that state in the future to keep other things we might run into very subtle bugs here.

Can you look into ^ in a separate PR?